client/src/interface/Profile.ts

```diff
@@ -0,0 +1,21 @@
+import { User } from './User';
+
+type Gender = 'male' | 'female';
+
+export interface Profile {
+  _id: string;
+  firstName: string;
+  lastName: string;
+  dob: string;
+  user: User;
```

Comment:
After some thought, I don't think we will need this property `user: User`. And here is why, possible operations on a profile object is
##### Create a profile
- the server can get the user data from the token supplied with the request
##### Update a profile
- this can be done with the profile ID
##### Get profile/profiles
- profiles can be sorted/searched by name, address, availability.
- and if a registered user wants to retrieve their profile, again the server already knows who the user is.
it seems there is never a need for the user property. please check to see if removing affects your implementation and let me know. if it doesn't I think we can conveniently remove it from the profile interface

Reply:
I agree. Ideally, this change would be coupled with the exclusion of `User.userId` field in API responses 👍🏼